Update xdg-mime helper

MoritzKn opened the pull request:
## Description

This will respect the directory `/usr/local/share/applications/` when fetching mime infos.

## TODOs:
<!-- Just check off what what we know been done so far. We can help you with this stuff. -->
- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

krader1961 commented on share/functions/__fish_print_xdg_applications_directories.fish:
```diff
@@ -0,0 +1,10 @@
+        set search_path $search_path /usr/local/share/applications
+    end
+
+    for p in $search_path
```
FWIW, the fishy way to do this is `printf '%s\n' $search_path`.

MoritzKn replied:
I now use the loop to check that the directories exist. So I kept it.

krader1961 commented on share/functions/__fish_print_xdg_applications_directories.fish:
```diff
@@ -0,0 +1,10 @@
+function __fish_print_xdg_applications_directories --description 'Print directories where desktop files are stored'
+    set -l search_path ~/.local/share/applications /usr/share/applications
+    if test -d /usr/local/share/applications
```
Why does this directory need an explicit check for existence but not the other two? For example, my ubuntu system has */usr/share/applications* but does not have a *~/.local/share/applications* directory.

MoritzKn replied:
True this should probably be done for all paths. `__fish_print_xdg_mimetypes` redirects finds stderr to `/dev/null` but `__fish_print_xdg_mimeapps` doesn't.  This could course error massages in the output.

krader1961 commented on share/functions/__fish_print_xdg_mimetypes.fish:
```diff
@@ -1,3 +1,3 @@
 function __fish_print_xdg_mimetypes --description 'Print XDG mime types'
-    cat ~/.local/share/applications/mimeinfo.cache /usr/share/applications/mimeinfo.cache ^/dev/null | string match -v '[MIME Cache]' | string replace = \t
+    cat {(__fish_print_xdg_applications_directories)}/mimeinfo.cache ^/dev/null | string match -v '[MIME Cache]' | string replace = \t
```
The braces aren't necessary.

MoritzKn commented:
I adjusted my changes in according to the remarks. 

I also checked the XDG spec and updated the code to actually comply with it.
Relevant parts of the [XDG base directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables):

> `$XDG_DATA_HOME` defines the base directory relative to which user specific data files should be stored. If `$XDG_DATA_HOME` is either not set or empty, a default equal to `$HOME/.local/share` should be used. 

>  `$XDG_DATA_DIRS` defines the preference-ordered set of base directories to search for data files in addition to the `$XDG_DATA_HOME` base directory. The directories in `$XDG_DATA_DIRS` should be seperated with a colon ':'.
> 
> If `$XDG_DATA_DIRS` is either not set or empty, a value equal to `/usr/local/share/:/usr/share/` should be used. 

>  Other specifications may reference this specification by specifying the location of a data file as $XDG_DATA_DIRS/subdir/filename. This implies that: 
> [...]
> Lookups of the data file should search for ./subdir/filename relative to all base directories specified by $XDG_DATA_HOME and $XDG_DATA_DIRS . If an environment variable is either not set or empty, its default value as defined by this specification should be used instead. 

Relevant parts of the [XDG desktop entry specification](https://specifications.freedesktop.org/desktop-entry-spec/latest/ape.html):
> To determine the ID of a desktop file, make its full path relative to the `$XDG_DATA_DIRS` component in which the desktop file is installed, remove the "applications/" prefix, and turn '/' into '-'. 

And two other questions:
* What is with the `po/*.po` files? Those reference the changed and removed functions and their descriptions. How would I go about updating those?
* Should I squash the commits?

krader1961 commented on share/functions/__fish_print_xdg_mimetypes.fish:
```diff
@@ -1,3 +1,3 @@
 function __fish_print_xdg_mimetypes --description 'Print XDG mime types'
-    cat {(__fish_print_xdg_applications_directories)}/mimeinfo.cache ^/dev/null | string match -v '[MIME Cache]' | string replace = \t
+    cat {__fish_print_xdg_applications_directories}/mimeinfo.cache ^/dev/null | string match -v '[MIME Cache]' | string replace = \t
```
Did you test this? Because it won't work now :smile: "Braces" (sometimes called "curly brackets") are the curly variant. Parentheses are the round variant. You need the parentheses to indicate a subcommand should be run and its output captured. So you removed the wrong pair of punctuation marks. I'll fix it when I merge it. Just a reminder to always double-check that a seemingly trivial change doesn't introduce a problem.

krader1961 commented:
> What is with the po/*.po files?

Those are language translation files (known as "localization" and ["L10N"](https://en.wiktionary.org/wiki/L10n) files). In general you don't have to worry about those unless you happen to read/write one of the languages and want to help keep the translations in sync with the english messages.

> Should I squash the commits?

You can but as a general rule you don't need to. Whomever merges your PR (me in this case) will take care of it if that's the appropriate thing to do.

Squash merged as commit 5eb0b34da1 and 6e02ec83d1. Thx.
